src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -1,59 +1,75 @@
 package org.skypro.skyshop.basket;
-import java.util.Arrays;
-import java.util.List;
+
 import org.skypro.skyshop.product.Product;
 
+import java.util.*;
+
 public class ProductBasket {
-    private final Product[] products;
-    private int size;
+    private Map<String, List<Product>> products;
 
-    public ProductBasket(int basket) {
-        products = new Product[basket];
-        size = 0;
+    public ProductBasket() {
+        products = new HashMap<>();
     }
+
     public void addProduct(Product product) {
-        if (size < products.length) {
-            products[size] = product;
-            size++;
-        } else {
-            System.out.println("Не возможно добавить продукт");
-        }
+        String name = product.getName();
+        products.computeIfAbsent(name, k -> new ArrayList<>()).add(product);
     }
+
+    public List<Product> removeProductByName(String name) {
+        List<Product> removedProduct = products.remove(name);
+        return removedProduct != null ? removedProduct : new ArrayList<>();
```

Comment:
лучше вернуть реально пустой список, а не массив из 10 элементов
```suggestion
        return removedProduct != null ? removedProduct : Collections.emptyList();
```